File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/champion/BlockDao.java
package com.legue.axel.lolsummonertool.database.dao.champion;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.legue.axel.lolsummonertool.database.model.champion.Block;

import java.util.List;

@Dao
public interface BlockDao {

    @Query("SELECT * FROM Blocks ORDER BY id")
    LiveData<List<Block>> getBlocks();

    @Query("SELECT * FROM blocks WHERE id = :blockId")
    LiveData<Block> getBlockById(int blockId);

    @Insert
    void insertBlock(Block block);

    @Delete
    void deleteBlock(Block ingredient);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateBlock(Block block);

}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/champion/BlockDao.kt
package com.legue.axel.lolsummonertool.database.dao.champion

import androidx.lifecycle.LiveData
import androidx.room.*
import com.legue.axel.lolsummonertool.database.model.champion.Block

@Dao
interface BlockDao {
    @Query("SELECT * FROM Blocks ORDER BY id")
    fun getBlocks(): LiveData<List<Block>>

    @Query("SELECT * FROM blocks WHERE id = :blockId")
    fun getBlockById(blockId: Int): LiveData<Block>

    @Insert
    fun insertBlock(block: Block)

    @Delete
    fun deleteBlock(ingredient: Block)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateBlock(block: Block)
}